### DISCUSSION TITLE
[Support]: Can't recognize usps logo, but will recognize amazon and fedex

### DISCUSSION THREAD

**Discussion by glenniana18** [blakeblackshear/frigate#23891] [General Support] [answered]
### Describe the problem you are having

Hi there!
I use Frigate+.  No matter what I do, my gate camera will not recognize the USPS logo.  It recognizes the Amazon and USPS trucks and their logos when they drive by, and will also recognize the USPS vehicle itself, but it just won't see the USPS logo.  I have "usps" listed in my tracked objects for that camera along with the others.  I have also submitted several snapshots of the usps logo to frigate+ and updated my model.  I am not sure what else to do?

Attaching my config.  Camera in question is called "gate".

I also seem to be getting lots of errors in my logs, but not sure what they are related to.  All my cameras are wired except the gate camera.

Any help you could provide would be greatly appreciated!

Note:  I am having an impossible time uploading my logs and config to this thread... I keep getting errors no matter what I do.  Seems like something's going on on the server side.  I will come back and try again in a few hours!

### Version

 0.17.1-416a9b7

### Frigate config file

```yaml
Will attach
```

### Relevant Frigate log output

```shell
Will attach
```

### Relevant go2rtc log output

```shell
Will attach
```

### FFprobe output from your camera

```shell
Will attach
```

### Install method

Proxmox via Docker

### docker-compose file or Docker CLI command

```yaml
Will attach
```

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Reolink doorbell

**Comment by inkeep-ai-assistant** [comment]
The USPS logo detection issue you're experiencing is likely related to the fine-tuning process with Frigate+. Since you mentioned that Amazon and FedEx logos are being recognized but USPS is not, here are some things to consider:

**Continue the Frigate+ fine-tuning workflow:**
- Frigate+ includes a fine-tuning workflow where you can upload images from your specific camera and correct false positives/negatives to create scene-specific models [(1)](https://docs.frigate.video/configuration/object_detectors)
- Keep submitting snapshots of the USPS logo from your gate camera, ensuring you're capturing various angles, lighting conditions, and distances
- After submitting annotations, make sure you've downloaded and applied the updated model to your Frigate configuration

**Factors that may affect detection:**
- Your gate camera is wireless, which can cause less reliable streams and potentially lost video data compared to wired cameras [(2)](https://docs.frigate.video/frigate/hardware)
- The USPS logo may appear at different sizes, angles, or lighting conditions than the training data currently covers

**Recommendations:**
1. Submit more diverse USPS logo images - different times of day, weather conditions, and positions in the frame
2. Verify your model has been updated after submitting annotations by checking that the model ID in your config matches the latest Frigate+ model
3. Consider the wireless connection - WiFi cameras can have stream reliability issues that may affect detection consistency [(2)](https://docs.frigate.video/frigate/hardware)

Since I don't have specific documentation about USPS logo detection requirements or thresholds, you may want to reach out through the Frigate community for additional support: [GitHub](https://github.com/blakeblackshear/frigate)

Would you like me to provide a general answer about object detection tuning strategies?

📚 **Sources:**
1. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
2. [Recommended hardware](https://docs.frigate.video/frigate/hardware)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by hawkeye217** [comment]
You should post your full config.

**Reply by glenniana18** [reply]
Here is my full config and logs.  Thank you for your help!!

[frigate config (3).txt](https://github.com/user-attachments/files/30637632/frigate.config.3.txt)
[frigate logs.txt](https://github.com/user-attachments/files/30637637/frigate.logs.txt)

**Reply by hawkeye217** [reply]
In addition to the AI's suggestions above, you should try lowering your `min_score` value for `usps`, eg:

```
objects: #may have to edit these
  filters:
    deer:
      min_score: 0.35 #any detection below this number will be ignored as a false positive
      threshold: 0.8 #a true positive is an object with a score >= this
    ...
    car:
      min_score: .65
      threshold: .85
    usps:
      min_score: 0.5      # lowered from the default
```

**Reply by glenniana18** [reply]
I've already submitted several images of the usps vehicle and they all look very similar to when the truck drives by every day - but it's still not catching it.  And yes i've verified that I'm running the updated model after those submissions were sent.

I will try lowering the threshold and see if that helps.  Thank you for your help!

**Reply by glenniana18** [reply]
Well that didn't work.  I lowered the threshold to 0.3 and still no detection.  I'm attaching the picture here... is this the type of logo y'all have on file for usps? Is this large enough for frigate to detect?

<img width="577" height="433" alt="usps" src="https://github.com/user-attachments/assets/eb84c85c-5019-4f82-ad6d-f7fee96e9097" />

**Reply by NickM-27** [reply]
Yes, this camera is fairly overexposed. Have you tried replaying the clip to see what is detected? https://docs.frigate.video/troubleshooting/dummy-camera

**Reply by glenniana18** [reply]
Oh this is cool!  I didn't know you could do this type of test.  I just did it and the debug view is showing that it detected the logo with a very high score!  (Although I'm not sure why the bounding box is around the whole car and not just the logo?)

<img width="894" height="659" alt="image" src="https://github.com/user-attachments/assets/4d2517f9-4794-457b-9431-7d293bd5759c" />

But when I go to the actual snapshot and play the clip, it shows it detected the car but does not list the usps logo as being detected:

<img width="976" height="629" alt="image" src="https://github.com/user-attachments/assets/d857fbd1-d13d-4f49-9900-8c3cb5901dca" />

And when I submit these images to Frigate+, it doesn't show the usps logo as being detected there either.  And I'm not getting the "usps in street" alert I set up for Home Assistant.  What am I missing?

**Reply by NickM-27** [reply]
The logos are used to apply the sub label to the object. The debug view will show it as USPS because that's what it named.

You will also see a bounding box just around the logo in the exact frames it is detected. 

Your screenshots make it look like you're comparing two different cameras. You only enabled USPS on the gate camera, none of the others

**Reply by glenniana18** [reply]
This is all from one camera - the gate camera. I only have one gate camera and it is mounted permanently.  

Let me compare this to what happens when Fedex comes.  Here is a screenshot from Fedex... it shows that Fedex itself was detected, and the car is recognized as fedex.

<img width="997" height="638" alt="image" src="https://github.com/user-attachments/assets/7c0f2007-8885-45e1-a219-9a25cea05df1" />

Compare that to USPS, which is only ever detected as "car" and never "USPS"
<img width="972" height="637" alt="image" src="https://github.com/user-attachments/assets/7f5388cf-f5f2-40b5-a9e9-e2b373381629" />

Why are these two things different?

**Reply by NickM-27** [reply]
The debug view is showing the sub label, so something is incorrect about the gate cameras config causing issues. What config did you use for the dummy camera testing?

**Reply by hawkeye217** [reply]
So just to be clear, did you set `threshold` for `usps` or `min_score`? Because `threshold` has no effect on attribute labels - you should be setting `min_score: 0.5` (or lower) for `usps`.

Another thing to note is that your detect resolution and distance of the camera to the van is likely causing the logo to be quite small. The FexEx logo was detected as 1232px, which is only a ~35x~35px square. The USPS logo is much smaller. What is your `detect` stream resolution?

**Reply by glenniana18** [reply]
[dummy_config.txt](https://github.com/user-attachments/files/30684700/dummy_config.txt)
Here's the dummy camera config.  I used what the docs told me, and then added the rest of the config from my gate camera to try to match it as close as possible.

**Reply by glenniana18** [reply]
I set min_score for usps to 0.3.  But frigate saw it with a score of 93%, so that shouldn't be the issue, right?

My detect stream resolution is 640x480.  I used these same settings in the dummy camera to try to replicate it, but it tracked it there.

**Reply by hawkeye217** [reply]
Does anything change if you increase the `detect` width and height in the dummy camera to 1280x960?

**Reply by glenniana18** [reply]
Just tried that, and it still sees USPS with a ~93% score in the debug view.

**Reply by hawkeye217** [reply]
Should be fine, then.

**Reply by NickM-27** [reply]
The 92% score is for the `car` object, just the name comes from USPS because that was detected in a previous frame. You'd need to see the frame that the bounding box is around the logo to know the score there.

Can you show us the event screenshots for the dummy camera specifically? The screenshot shown before looks like it lasted for a long time, even after the clip was run (based on the timestamp in the screenshot).

**Reply by NickM-27** [reply]
Your dummy camera has recording disabled so it won't exist, most likely the clip looping multiple times is allowing more chances for detection. It would be good to see a confirmation with recording enabled.

**Reply by hawkeye217** [reply]
Right. I would also suggest adding some "padding" time before and after the detection in the looping clip so that the detection is lost and regained each loop.

**Reply by glenniana18** [reply]
Oh ok, thanks for explaining the 92%.

I'm not sure what you mean by "event screenshots" for the dummy camera.  The dummy camera is set up with "record" and "snapshots" both false per the docs - should I change that?

And yeah, it lasts for a long time... the car drives from the right side of the screen to the left, stops at all the mailboxes on the way, turns around, and then comes back down the street from left to right.  The logo is in view a LOT.

**Reply by glenniana18** [reply]
got it.  Thanks guys!  Let me add more padding, turn recording on for the dummy camera, and report back.

**Reply by glenniana18** [reply]
Ok, added padding and turned on recording and snapshots.

Car was recognized as USPS in the dummy camera:

<img width="990" height="490" alt="image" src="https://github.com/user-attachments/assets/643df4a7-aa74-4c02-bf2b-1c803f24e635" />

<img width="1001" height="645" alt="image" src="https://github.com/user-attachments/assets/73bf8105-4b12-42d1-86e4-73b213ad57ae" />

This is with the new higher-resolution on the dummy camera's detect stream.  I changed it back to 640x480 to see if that would change anything, and now i'm getting a bunch of errors in my logs for the dummy camera: "Unable to read frames from ffmpeg process." Not sure what that's about...

**Reply by hawkeye217** [reply, marked as answer]
In the dummy camera loop, the USPS logo is only 962px in area (representing a ~31x31px square) with the higher resolution. If you lowered it back to 640x480, the logo area would only potentially be ~241px, which is represented by a ~15x15 square. This may be too small to be accurately and consistently recognized.

I'd suggest increasing your `detect` width and height on the real camera and seeing what happens when the mail comes today.

**Reply by glenniana18** [reply]
You called it!  The camera is a Reolink doorbell, so it has a "sub" stream that is 640x480 at 7 fps, and an "ext" stream that is 896x672 at 20 fps.  These settings are fixed.  I switched to the "ext" stream as my detect stream and it caught the USPS logo today!  Woo hoo!

Do you think keeping the 20 fps stream will work for detection without overloading things?  I remember reading that the detect stream is supposed to be a lower fps, but it looks like this is really my only option if I want to get the USPS notifications.

Thank you so much for the help - I truly appreciate it!

**Reply by hawkeye217** [reply]
Great! Glad to hear it's working now.

I'd suggest just setting the `fps` value in your `detect` config to 5. This is the recommended value, and it does *not* need to match the output of your camera.

```
detect:
  width: 896
  height: 672
  fps: 5
```

I wrote up a section of the docs for the next version of Frigate with more details on the recommended detect `fps` value [here](https://docs-dev.frigate.video/frigate/camera_setup/#choosing-a-detect-frame-rate), but the information still applies to the current version.

**Reply by glenniana18** [reply]
Thank you!  I always thought these 3 settings had to match the camera's output.  Great info.
